csharp-design-patterns/AggregateDesign/AggregateDesign/DataAccess/OrdersRepository.cs

```diff
@@ -0,0 +1,57 @@
+﻿using AggregateDesign.Domain;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+
+namespace AggregateDesign.DataAccess;
+
+public class OrdersRepository : IOrdersRepository
+{
+    private readonly OrdersDbContext _dbContext;
+    private readonly IMapper _mapper;
+
+    public OrdersRepository(OrdersDbContext dbContext, IMapper mapper)
+    {
+        _dbContext = dbContext;
+        _mapper = mapper;
+    }
+
+    public async Task<Order?> GetByIdAsync(long id)
+    {
+        var dataModel = await _dbContext.orders
+            .Include(x => x.Items)
+            .FirstAsync(x => x.OrderId == id);
+
+        if (dataModel == null)
```

Comment:
This better be (is null) following best practices. We have an article on why if you are curious :D

Reply:
done